cuppa-core/src/main/java/org/forgerock/cuppa/reporters/DefaultReporter.java

```diff
@@ -0,0 +1,118 @@
+package org.forgerock.cuppa.reporters;
+
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.forgerock.cuppa.Cuppa;
+import org.forgerock.cuppa.Reporter;
+
+/**
+ * An RSpec-like reporter.
+ */
+public final class DefaultReporter implements Reporter {
+    private final PrintStream stream;
+    private final List<TestFailure> failures = new ArrayList<>();
+    private final Deque<String> blockStack = new ArrayDeque<>();
+    private int passed;
+    private int failed;
+    private int pending;
+
+    /**
+     * Constructs a reporter that writes to standard out.
+     */
+    public DefaultReporter() {
+        this(System.out);
+    }
+
+    /**
+     * Constructs a reporter that writes to the specified stream.
+     *
+     * @param stream A stream to write to.
+     */
+    public DefaultReporter(OutputStream stream) {
+        this.stream = new PrintStream(stream);
+    }
+
+    @Override
+    public void start() {
+        stream.println();
+    }
+
+    @Override
+    public void end() {
+        stream.println();
+        stream.println();
+        stream.println("  " + passed + " passing");
+        if (failed > 0) {
+            stream.println("  " + failed + " failing");
+        }
+        if (pending > 0) {
+            stream.println("  " + pending + " pending");
+        }
```

Comment:
Will need to add "skipped" once rebased on the supporting skipping tests branch.